drivers/hmis/app/graphql/mutations/save_assessment.rb

```diff
@@ -1,36 +1,54 @@
 module Mutations
   class SaveAssessment < BaseMutation
-    argument :assessment_id, ID, required: true
-    argument :values, Types::JsonObject, required: true
-    argument :in_progress, Boolean, required: false
+    description 'Create/Save assessment as work-in-progress'
+
+    argument :assessment_id, ID, 'Required if updating an existing assessment', required: false
+    argument :enrollment_id, ID, 'Required if saving a new assessment', required: false
+    argument :form_definition_id, ID, 'Required if saving a new assessment', required: false
```

Comment:
Not sure this is possible, but it'd be nice to have something like:
```suggestion
    argument :assessment_id, ID, 'Required if updating an existing assessment', required: persisted?
    argument :enrollment_id, ID, 'Required if saving a new assessment', required: !persisted?
    argument :form_definition_id, ID, 'Required if saving a new assessment', required: !persisted?
```

Reply:
I've seen something like that with a lambda, but not sure that's the right approach for a mutation.

Reply:
We could put them in separate types and use ruby `oneOf` validation https://graphql-ruby.org/type_definitions/input_objects#oneof but I don't know if that's worth it. At the end of the day it is still a runtime check, none of the fields are required in the schema.